engine-rest/engine-rest/src/test/java/org/camunda/bpm/engine/rest/ProcessInstanceRestServiceInteractionTest.java

```diff
@@ -3257,21 +3265,69 @@ public void testSetRetriesByProcessAsync() {
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).processInstanceQuery(null);
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).dueDate(null);
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).executeAsync();
+    verifyNoMoreInteractions(mockSetJobRetriesByProcessAsyncBuilder);
+  }
+
+  @Test
+  public void testSetRetriesByProcessAsyncWithDueDate() {
+    List<String> ids = Arrays.asList(MockProvider.EXAMPLE_PROCESS_INSTANCE_ID);
+
+    Map<String, Object> messageBodyJson = new HashMap<>();
+    messageBodyJson.put("processInstances", ids);
+    messageBodyJson.put(RETRIES, MockProvider.EXAMPLE_JOB_RETRIES);
+    Date newDueDate = new Date(0);
+    messageBodyJson.put("dueDate", newDueDate);
+
+    Response response = given()
+        .contentType(ContentType.JSON).body(messageBodyJson)
+        .then().expect()
+        .statusCode(Status.OK.getStatusCode())
+        .when().post(SET_JOB_RETRIES_ASYNC_URL);
+
+    verifyBatchJson(response.asString());
+
+    verify(mockManagementService, times(1)).setJobRetriesByProcessAsync(eq(MockProvider.EXAMPLE_JOB_RETRIES));
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).processInstanceIds(Arrays.asList(MockProvider.EXAMPLE_PROCESS_INSTANCE_ID));
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).processInstanceQuery(null);
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).dueDate(eq(newDueDate));
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).executeAsync();
+    verifyNoMoreInteractions(mockSetJobRetriesByProcessAsyncBuilder);
   }
 
   @Test
   public void testSetRetriesByProcessAsyncWithQuery() {
-    Batch batchEntity = MockProvider.createMockBatch();
-    when(mockManagementService.setJobRetriesAsync(
-        any(),
-        Mockito.<ProcessInstanceQuery>any(),
-        anyInt())
-    ).thenReturn(batchEntity);
+    Map<String, Object> messageBodyJson = new HashMap<>();
+    messageBodyJson.put(RETRIES, MockProvider.EXAMPLE_JOB_RETRIES);
+    HistoricProcessInstanceQueryDto query = new HistoricProcessInstanceQueryDto();
+    messageBodyJson.put("processInstanceQuery", query);
+    Date newDueDate = new Date(0);
+    messageBodyJson.put("dueDate", newDueDate);
 
-    Map<String, Object> messageBodyJson = new HashMap<String, Object>();
-    messageBodyJson.put(RETRIES, 5);
+    Response response = given()
+        .contentType(ContentType.JSON).body(messageBodyJson)
+        .then().expect()
+        .statusCode(Status.OK.getStatusCode())
+        .when().post(SET_JOB_RETRIES_ASYNC_URL);
+
+    verifyBatchJson(response.asString());
+
+    verify(mockManagementService, times(1)).setJobRetriesByProcessAsync(eq(MockProvider.EXAMPLE_JOB_RETRIES));
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).processInstanceIds(null);
+    verify(mockSetJobRetriesByProcessAsyncBuilder, times(1)).processInstanceQuery(null);
```

Comment:
Maybe I am missing something here, but shouldn't `processInstanceQuery` have been called on the builder since we added a `processInstanceQuery` parameter to the call? 🤔

Reply:
Jep, you are right I missed this. The test works because the `runtimeServiceMock` does not return a process instance query in this setup.